src/zimcheck.cpp

```diff
@@ -361,13 +356,14 @@ void test_articles(const zim::File& f, ErrorLogger& reporter, ProgressBar progre
             if (it->getMimeType() != "text/html")
                 continue;
 
-            auto links = getDependencies(it->getPage());
-            for (auto &link: links)
+            if (url_check == false)
+                links = generic_getLinks(it->getPage());
```

Comment:
Maybe move this in line 289 with :
```
if (url_check || url_check_external) {
    links = generic_getLinks(it->getData())
}
```
(`getPage` will be removed in next api)

Reply:
done